### DISCUSSION TITLE
IMU not showing using dump

### DISCUSSION THREAD

**Discussion by sam-tj** [ifm/ifm3d#417] [General]
I am trying to check for imu information from port 6 but for my device, port 6 is not available. Do I have to enable it from some option or does it need additional configuration? 

My hardware config is:
- OPV800
- O3R225

The plan is to inspect the imu data and see if it is usable for image stability. 

Thanks in advance.

**Comment by lola-masson** [comment]
Hi @sam-tj, 

We currently do not stream the imu data unfortunately. This is on the roadmap for a couple firmware versions down the road (I don't have an exact date for you right now).

Sorry about that!

**Reply by sam-tj** [reply]
Thanks for your reply.

In this case, what would you suggest for imu integration?

I was looking forward to use the imu data from base library (c++ or python) but that is not possible.

**Reply by desengph** [reply]
Hi @sam-tj 
as @lola-masson has stated before, the IMU is integrated on our system, but is currently not activated for a customer use case, as there is no public interface to access its data.
We're working on providing a solution to the data access problem in the next FW versions.

For your information:
We are using a TDK IIM-42652 IMU on our system. It's a 6-axis IMUs that incorporate a 3-axis gyroscope and 3-axis accelerometer.
Can you please verify if such this measurement unit is sufficient for your application based on its specification and data output?

Can you please specify the application use case in more detail: image stability?
If this information may be too detailed to share on GH, please let us know via email and we can keep the conversation alive via: support.robotics@ifm.com

BR
Philipp

**Reply by sam-tj** [reply]
Hi @desengph ,

Thanks for your reply. 

I would be working on a project on ros2 with visual mapping. The IMU data is required for the mapping package to improve stability. Thus was hoping that something is possible with the hardware directly. Else I would attach an external imu and try syncing both.

**Reply by desengph** [reply]
Hi @sam-tj, thank you for the clarification.

Since the update to make the IMU data available will require additional work from our side, it will take a bit of time. 
We'll let you know once this becomes available. For now, can you please use an external IMU source as suggested?
